jwala-services/src/main/java/com/cerner/jwala/service/app/impl/ApplicationServiceImpl.java

```diff
@@ -128,7 +132,42 @@ public Application getApplication(final String name) {
     public Application updateApplication(UpdateApplicationRequest updateApplicationRequest, User anUpdatingUser) {
         updateApplicationRequest.validate();
 
-        return applicationPersistenceService.updateApplication(updateApplicationRequest);
+        final Application application = applicationPersistenceService.updateApplication(updateApplicationRequest);
+        updateApplicationWarMetaData(updateApplicationRequest, application);
+        return application;
+    }
+
+    private void updateApplicationWarMetaData(UpdateApplicationRequest updateApplicationRequest, Application application) {
+        final String appWarName = application.getWarName();
+        if (!appWarName.isEmpty()) {
+            final String appName = application.getName();
+            try {
+                String originalJsonMetaData = groupPersistenceService.getGroupAppResourceTemplateMetaData(application.getGroup().getName(), appWarName);
+                ResourceTemplateMetaData originalMetaData = resourceService.getMetaData(originalJsonMetaData);
+                ResourceTemplateMetaData updateMetaData = new ResourceTemplateMetaData(
+                        originalMetaData.getTemplateName(),
+                        originalMetaData.getContentType(),
+                        originalMetaData.getDeployFileName(),
+                        originalMetaData.getDeployPath(),
+                        originalMetaData.getEntity(),
+                        updateApplicationRequest.isUnpackWar(),
```

Comment:
The unpackWar flag is the only meta data attribute that is shared by the Web App. The other Web App attributes (application name, application context path, is the application secure, and the load balance attribute) don't map to any meta data attributes. Technically, the application name and group details could be mapped to the entity attributes of the meta data, but those are no longer used for deployment or identifying the resource.

Also, this code is only called when the Web App is updated - not when the meta data is updated. So this will map the "Unpack War" attribute of the application to the "unpack" attribute of the meta data for the application war resource.